src/basicio.cpp

```diff
@@ -1919,44 +1872,24 @@ namespace Exiv2 {
         src.close();
     }
 
-#if defined(_MSC_VER)
     int RemoteIo::seek( int64_t offset, Position pos )
     {
         assert(p_->isMalloced_);
-        uint64_t newIdx = 0;
+        int64_t newIdx = 0;
 
         switch (pos) {
             case BasicIo::cur: newIdx = p_->idx_ + offset; break;
             case BasicIo::beg: newIdx = offset; break;
             case BasicIo::end: newIdx = p_->size_ + offset; break;
         }
 
-        if ( /*newIdx < 0 || */ newIdx > static_cast<uint64_t>(p_->size_) ) return 1;
-        p_->idx_ = static_cast<long>(newIdx);   //not very sure about this. need more test!!    - note by Shawn  fly2xj@gmail.com //TODO
-        p_->eof_ = false;
-        return 0;
-    }
-#else
-    int RemoteIo::seek(long offset, Position pos)
-    {
-        assert(p_->isMalloced_);
-        long newIdx = 0;
-
-        switch (pos) {
-            case BasicIo::cur: newIdx = p_->idx_ + offset; break;
-            case BasicIo::beg: newIdx = offset; break;
-            case BasicIo::end: newIdx = p_->size_ + offset; break;
-        }
+        if (newIdx < 0 || static_cast<size_t>(newIdx) > p_->size_)
+            return 1;
 
-        // #1198.  Don't return 1 when asked to seek past EOF.  Stay calm and set eof_
-        // if (newIdx < 0 || newIdx > (long) p_->size_) return 1;
-        p_->idx_ = newIdx;
-        p_->eof_ = newIdx > static_cast<long>(p_->size_);
-        if (p_->idx_ > static_cast<long>(p_->size_))
-            p_->idx_ = static_cast<long>(p_->size_);
+        p_->idx_ = static_cast<long>(newIdx);
+        p_->eof_ = static_cast<size_t>(newIdx) > p_->size_;
```

Comment:
```suggestion
        p_->eof_ = false;
```

Reply:
On second thoughts, I think my suggestion is wrong. I noticed that your assignment `p_->eof_ = static_cast<size_t>(newIdx) > p_->size_` was equivalent to `p_->eof_ = false` due to the early return. But it also doesn't make sense to change `p_->eof_` if `p_->idx_` doesn't change.

Reply:
I considered your suggestions and thanks to that I realised that MemIo and RemoteIo were sharing the same implementation for the `seek` method. Therefore, I have factored out that code and reuse it.

Reply:
Does fseek() set `eof_` in the standard C library?

Emulating `FILE*` semantics in BasicIo is a good design.  @piponazo, you'll remember I didn't do that with RemoteIo and got into trouble like billions of `fstat()` calls.

I don't like the early return.  As horrible as `goto`.  I'd only allow return on the initial and last statement of a function.  _(I've never met anybody who agrees with me about this.  That doesn't change my opinion!)_.

Reply:
`std::fseek` does not seems to set EOF in the stream, other operations like `std::fgetc` or `std::fread` sets EOF when one tries to read beyond the limits of the stream. After that, you can check the EOF status with `std::feof`. That's actually what we do in `FileIo`.

Regarding the multiple return calls, I do not have strong opinions about it. I normally would agree with you (early return or return at the end). However, even in the Cpp Core Guidelines, people have make a note about this "old rule":
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#nr2-dont-insist-to-have-only-a-single-return-statement-in-a-function

I think I will leave the seek implementations as they were before, because it was not the main intention of this PR to refactor that code. I do not think there is a major issue there (problably the RemoteIO class is not used too much). I just noticed that the implementations of `MemIo::seek` and `RemoteIo::seek` were almost similar and it could make sense to reuse the same code for both.